### PR TITLE
Fix #8857: Calendar doesn't need AJAX protection on setDate

### LINKED ISSUE #8857: Selenium: TimeoutException in complex mask with many update="@form"
### Describe the bug

Let's try this (maybe) difficult one. Running a Selenium test on Firefox featuring a complex mask with many update="@form" fails with TimeoutException.

It does not happen always so it might be a timing issue or race condition or concurrency problem. I have seen it with

org.primefaces.selenium.component.InputText
org.primefaces.selenium.component.Calendar
org.primefaces.selenium.component.InputTextarea

but interestingly NOT with org.primefaces.selenium.component.InputNumber

In case you cannot reproduce it, here is some more information what happens as far as I understand it.

The PrimeSelenium Guard class sets this indicator before executing the command:
"pfselenium.xhr = 'somethingJustNotNull';"

In waitUntilAjaxCompletes it checks among other stuff:
pfselenium.xhr == null

In the error case it seems that pfselenium.xhr is not reset to null like this debug output shows:

Guard#ajax; ajaxDebugInfo **before methode.invoke**: document.readyState=complete, !window.jQuery=false, jQuery.active=0, !window.PrimeFaces=false, PrimeFaces.ajax.Queue.isEmpty()=true, PrimeFaces.animationActive=false, !window.pfselenium=false, **pfselenium.xhr=somethingJustNotNull**, pfselenium.anyXhrStarted=null, pfselenium.navigating=false

Guard#ajax; ajaxDebugInfo **in error catch clause**: document.readyState=complete, !window.jQuery=false, jQuery.active=0, !window.PrimeFaces=false, PrimeFaces.ajax.Queue.isEmpty()=true, PrimeFaces.animationActive=false, !window.pfselenium=false, **pfselenium.xhr=somethingJustNotNull**, pfselenium.anyXhrStarted=null, pfselenium.navigating=false

Therefore the condition for "Ajax complete" is never fulfilled and a TimeoutException occurs.

### Reproducer

Will provide a pull request with an appropriate integration test (hopefully) producing the error.

### Expected behavior

Selenium test runs fine without TimeoutException.

### PrimeFaces edition

Community

### PrimeFaces version

11

### Theme

Any

### JSF implementation

MyFaces

### JSF version

2.2, 2.3

### Browser(s)

Firefox 100

### PR DESCRIPTION
**Pull request by melloware**

### CODE REVIEW

**Comment by tandraschko** [comment]
Can you explain? I thought this method fires the behavior event

**Comment by melloware** [comment]
I looked in Calendar and it does not fire any event.  The `dateSelect` event is only fired when a date is picked from the popup panel.  It just does this...

```js
 setDate: function(date) {
        this.jqEl.datetimepicker('setDate', date);
    }
```

Setting the date on the jq picker.

**Comment by melloware** [comment]
I will admit since we poured all our effort into DatePicker there are no Calendar tests at all so this has probably never worked.
